### PR TITLE
[New Rules] Potential PowerShell Pass-the-Hash/Relay Script

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Detects PowerShell scripts that can execute pass-the-hash (PtH) attacks, intercept and relay NTLM challenges, and carry out other man-in-the-middle (MitM) attacks.

Some related tooling:

* https://github.com/Kevin-Robertson/Invoke-TheHash/blob/master/Invoke-WMIExec.ps1
* https://github.com/Kevin-Robertson/Invoke-TheHash/blob/master/Invoke-SMBExec.ps1
* https://github.com/dafthack/Check-LocalAdminHash/blob/master/Check-LocalAdminHash.ps1
* https://github.com/nettitude/PoshC2/blob/master/resources/modules/Invoke-Tater.ps1
* https://github.com/Kevin-Robertson/Inveigh/blob/master/Inveigh.ps1

### CODE REVIEW

**Review by Samirbous — APPROVED** [review]
nice!

File: rules/windows/credential_access_posh_relay_tools.toml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+severity = "medium"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Credential Access", "Resources: Investigation Guide", "Data Source: PowerShell Logs"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.category:process and host.os.type:windows and
+  powershell.file.script_block_text : (
+    ("NTLMSSPNegotiate" and ("NegotiateSMB" or "NegotiateSMB2")) or
+    "4E544C4D53535000" or
```
Comment (w0rk3r) [inline rules/windows/credential_access_posh_relay_tools.toml:56]: NTLMSSP in hex

File: rules/windows/credential_access_posh_relay_tools.toml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    "0x4e,0x54,0x4c,0x4d,0x53,0x53,0x50"
```
Comment (w0rk3r) [inline rules/windows/credential_access_posh_relay_tools.toml:57]: NTLMSSP in hex

File: rules/windows/credential_access_posh_relay_tools.toml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    "0x4e,0x54,0x20,0x4c,0x4d" or
```
Comment (w0rk3r) [inline rules/windows/credential_access_posh_relay_tools.toml:58]: NT LM 0.12 in hex

File: rules/windows/credential_access_posh_relay_tools.toml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    "0x53,0x4d,0x42,0x20,0x32" or
```
Comment (w0rk3r) [inline rules/windows/credential_access_posh_relay_tools.toml:59]: SMB 2.002 in hex

File: rules/windows/credential_access_posh_relay_tools.toml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    "0x81,0xbb,0x7a,0x36,0x44,0x98,0xf1,0x35,0xad,0x32,0x98,0xf0,0x38"
```
Comment (w0rk3r) [inline rules/windows/credential_access_posh_relay_tools.toml:60]: SMB named pipe UUID

File: rules/windows/credential_access_posh_relay_tools.toml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+reg add "hklm\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging" /v EnableScriptBlockLogging /t REG_DWORD /d 1
+```
+"""
+severity = "medium"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Credential Access", "Resources: Investigation Guide", "Data Source: PowerShell Logs"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.category:process and host.os.type:windows and
```
Comment (Aegrah) [inline rules/windows/credential_access_posh_relay_tools.toml:53]: There is no `event.type == "start"` or something in these events?

**Review by Aegrah — APPROVED** [review]
Nice!

File: rules/windows/credential_access_posh_relay_tools.toml (hunk 6 above)
Comment (w0rk3r) [reply rules/windows/credential_access_posh_relay_tools.toml:53]: Hmmm there is `event.type == "info"`, maybe makes sense to push this in a separated PR, as none of the other PowerShell rules include it

Comment (Aegrah) [reply rules/windows/credential_access_posh_relay_tools.toml:53]: Think that would be good; especially giving the broad search of powershell text fields.

Comment (w0rk3r) [reply rules/windows/credential_access_posh_relay_tools.toml:53]: Actually, I think we may not need to do that, as we specifically use the `logs-windows.powershell*` index

**Pull request merged**